params/precompile_upgrade.go

```diff
@@ -149,18 +149,33 @@ func (c *ChainConfig) verifyPrecompileUpgrades() error {
 	return nil
 }
 
-// GetActivePrecompileConfig returns the most recent precompile config corresponding to [address].
+// verifyStateUpgrades checks [c.StateUpgrades] is well formed:
+// - the specified blockTimestamps must monotonically increase
+func (c *ChainConfig) verifyStateUpgrades() error {
+	var previousUpgradeTimestamp *big.Int
+	for i, upgrade := range c.StateUpgrades {
+		upgradeTimestamp := upgrade.BlockTimestamp
+		// Verify specified timestamps are strictly monotonically increasing.
+		if previousUpgradeTimestamp != nil && upgradeTimestamp.Cmp(previousUpgradeTimestamp) <= 0 {
```

Comment:
if we decided not to use state upgrades at genesis, should we check if timestamp != 0? cc @aaronbuchwald

Reply:
Ya I think we should add this check since we are not allowing them in the genesis